File: Patrones/Estructurales/DAO/Java/src/UsuarioDAO.java
import java.util.List;

public interface UsuarioDAO {
    Usuario obtenerUsuarioPorId(int id);
    List<Usuario> obtenerTodosLosUsuarios();
    void insertarUsuario(Usuario usuario);
    void actualizarUsuario(Usuario usuario);
    void eliminarUsuario(int id);
}

File: Patrones/Estructurales/DAO/Python/UsuarioDAO.py
from abc import ABC, abstractmethod

class UsuarioDAO(ABC):
    
    @abstractmethod
    def obtenerUsuarioPorId(self, id):
        pass

    @abstractmethod
    def obtenerTodosLosUsuarios(self):
        pass

    @abstractmethod
    def insertarUsuario(self, usuario):
        pass

    @abstractmethod
    def actualizarUsuario(self, usuario):
        pass

    @abstractmethod
    def eliminarUsuario(self, id):
        pass